### PR TITLE
Crafting system

### PR DESCRIPTION
**Pull request by Hemat02**
Please check functionality. Crafting system fully implemented.

### CODE REVIEW

File: Assets/Scripts/Inventory/Crafting/AdranalinePotion.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class AdranalinePotion : MonoBehaviour
```
Comment (TK-786) [inline Assets/Scripts/Inventory/Crafting/AdranalinePotion.cs:5]: Adrenaline*

File: Assets/Scripts/Inventory/InventoryManager.cs [hunk 2]
```diff
@@ -7,46 +7,119 @@ public class InventoryManager : MonoBehaviour
...
     private bool menuOpen;
     public ItemSlot[] itemSlots;
+    public ItemSlot[] craftingSlots;
+    public ItemSlot craftingSlot;
     private PickUpScript pickUpScript;
 
+    public GameObject DescriptionPanel;
+    public GameObject CraftingPanel;
+
+    public CraftingManager craftingManager;
```
Comment (TK-786) [inline Assets/Scripts/Inventory/InventoryManager.cs:17]: I don't think it needs to be exposed @Hemat02?

File: Assets/Scripts/Inventory/InventoryManager.cs [hunk 3]
```diff
@@ -7,46 +7,119 @@ public class InventoryManager : MonoBehaviour
...
+        CraftingPanel.SetActive(false);
     }
 
     // Update is called once per frame
     void Update()
     {
-        if (Input.GetButtonDown("Inventory")){
-            Menu.SetActive(!menuOpen);  
+        // Raycasting to detect crafting table
+        if (Input.GetKeyDown(KeyCode.E))
```
Comment (TK-786) [inline Assets/Scripts/Inventory/InventoryManager.cs:40]: could be moved to pickup script potentially

File: Assets/Scripts/Monsters/Item.cs [hunk 4]
```diff
@@ -6,24 +6,25 @@
 public class Item : MonoBehaviour
 {
     [SerializeField]
-    private string itemName;
+    public string itemName;
```
Comment (TK-786) [inline Assets/Scripts/Monsters/Item.cs:9]: there are getters at the bottom of the file. Didn't need to be public

**Review by TK-786 — CHANGES_REQUESTED** [review]
Address comments, but seems good

**Review by HamidH3 — APPROVED** [review]
Works nice, well done.

**Review by TK-786 — APPROVED** [review]

**Pull request merged**